### ISSUE TITLE
http, meta and summary in graphite

### ISSUE THREAD

**Issue by namnn96** [sitespeedio/sitespeed.io#957]
Hi @soulgalore,

I just have another question regarding sitespeed.

I am running sitespeed with graphite and grafana on virtual box.
The first command I run, it generates 3 folder in graphite/whisper/<graphiteNamespace>: http, meta and summary. I also put it in crontab and it works fine for that.

However, when I run the same site, and create new nameSpace in graphite, it only gives me http, no matter if I run it as a -u <url> or -f <file>.

Personally, I find the summary more useful as it covers all information similar to the html in har folder.
Can you explain the structure and reason for this?

Many thanks.

P/s: I still cannot get the wpt work for my instance on aws. The error is "Invalid location"

**Comment by soulgalore** [comment]
Hey @namnn96 
the http namespace holds specific info per URL you test. So if you test 10 URLs all these individual metrics will be there. Use this if you want to follow the metrics on a specific page.

The summary is a summary for all the runs, so median values of all the tested pages.
Best
Peter

**Comment by namnn96** [comment]
Thank you for the quick reply.

The strange thing is that when i run this for the first time for the very
first site, i receive 3 folders http, meta and summary with all details.

Then the next time i run it, same site, different namespace in graphite, no
meta nor summary folders in graphite. And the http folder has very limited
info

What should i do?

**Comment by namnn96** [comment]
Actually i know the difference.

When I added the run command to crontab, it generates all 3 folders, not
one when running single command.

Also, for another site like google, it generates only 1 http, no meta nor
summary though i put it into crontab.
Can you explain what the crontab does aside from repeating the test after a
certain period of time?

Thank you

**Comment by namnn96** [comment]
Sorry Peter, I think I have one confusion that may need your explanation.

In the documentation on sitespeed homepage on creating dashboard with
graphite, grafana and sitespeed, I saw you used crontab.
What does the crontab do?

And am I right If I say the crontab is to run the sitespeed command for
every period of time in the background?
Is crontab usable for both docker run and ubuntu run?

I am still learning this. Thank you in advance

**Comment by soulgalore** [comment]
hey @namnn96 
the crontab is used to schedule a job, so for example make sure we test a URL every 5 minutes (or whatever interval you want to use). Yep you can use it to make that sitespeed.io runs at specific intervals. You can start docker containers or sitespeed directly.

Best
Peter

**Comment by namnn96** [comment]
I just came back to the module and I couldn't create the summary folder even though I ran the test for 9 times. 

Do you have any idea? What must-do to create summary folder successfully?

Thank you.

**Comment by soulgalore** [comment]
Hey @namnn96 ,
what do you mean by summary folder? The folder on disk with the HTML result or the view in Graphite? Do you use 3.x or the alpha of 4.0?

Best
Peter

**Comment by namnn96** [comment]
Hey @soulgalore ,

I mean the view in graphite. I can view everything in the HTML result, but it doesn't seem to send all data to graphite. When I open Graphite, under the destination namespace, there is only http folder. 

What do you mean by 3.x and alpha of 4.0? Please explain I am still learning this.

Thank you.

**Comment by soulgalore** [comment]
>What do you mean by 3.x and alpha of 4.0? 
3.x is the current release, 4.0 is the coming release. 

Usually the metrics is sent (but you can verify by add verbose logging). Can it be that Graphite is overloaded with metrics?

**Comment by namnn96** [comment]
Hi Peter,

I don't know how to check these tools versions.

Also when I add verbose to the command, it displays data for summary
folder, meta folder, http folder and then "info: Sending data to Graphite
host:localhost port:2003".
However, when I check the whisper folder, only http folder is there, I
really don't know why. Is that because graphite is full (but I have only
analysed 3 urls).

Thank you

**Comment by beenanner** [comment]
@namnn96 I would check your graphite logs to make sure your graphite instance isn't dropping metrics as they come in. Are you using the sitespeed.io graphite docker image or did you set it up independently? If the later then check out https://www.sitespeed.io/documentation/graphs/ at the bottom. You should set MAX_CREATES_PER_MINUTE = inf in your carbon.conf file and restart graphite.

**Comment by namnn96** [comment]
@beenanner  where can I find that log file?

I set up independently, no docker used. I changed the MAX_CREATES_PER_MINUTE to inf, restart server but it still miss the data in pagemetric and timing

**Comment by beenanner** [comment]
@namnn96 they should be in `/opt/graphite/storage/log` I believe.

**Comment by namnn96** [comment]
@beenanner I don't have graphite under opt.

Not sure if I set up correctly. I followed this tutorial to set up graphite https://www.digitalocean.com/community/tutorials/how-to-install-and-use-graphite-on-an-ubuntu-14-04-server

For sitespeed I installed npm and download sitespeed via npm. Just wonder if sitespeed will fetch the timing for every site or just some?

Thank you.

**Comment by beenanner** [comment]
@namnn96 I would guess they are in `/var/log/carbon`. I believe the default for sitespeed.io 3.x is that it sends all timing for every resource(s)/site(s), so depending on the size of the site(s) you are testing it could be a lot of data sent over to graphite.

**Comment by namnn96** [comment]
@beenanner I found a console.log under /var/log/carbon and there actually many unhandled errors there. I have posted a question on stackoverflow on this, but doesn't seem to get any answer. If you know the answer for  any of these questions, please leave an answer. Thank you.

[](http://stackoverflow.com/questions/38492805/understanding-how-graphite-sitespeed-io-work)

**Comment by namnn96** [comment]
Hi guys, 

I have played around with sitespeed for quite a long time now and lately there are a couple of issues.

+) I installed selenium 2.53.1 and firefox 46. It does not crash like firefox 47 but still cause some warning like Driver class not found. (it still runs fine)
+) The next strange thing is data sent to graphite, but they are all empty? I set max create per minute to inf but this still happens. (Earlier I tried on a server and it's fine, now that I install on another server.)
+) I also test via webpagetest. The issue is when running the sitespeed command with wpt option, it stops at running webpagetest, not crash, just doing nothing and after 1 hour I had to quit it.

I hope you guys know the issue, thanks in advance.

Best regards,
